src/hotspot/share/runtime/flags/jvmFlag.hpp

```diff
@@ -290,6 +291,23 @@ class JVMFlag {
 
   static const char* flag_error_str(JVMFlag::Error error);
 
+  // type checking
+#define CHECK_COMPATIBLE(type) \
+  case TYPE_##type: \
+    assert(sizeof(T) == sizeof(type) && \
+           std::is_integral<T>::value == std::is_integral<type>::value && \
+           std::is_signed  <T>::value == std::is_signed  <type>::value, "must be"); break;
```

Comment:
I'd prefer the `break` be on it's own source line, rather than after the assert form.